VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participant-waiting-room/private-consultation-participants/private-consultation-participants.component.ts

```diff
@@ -128,35 +128,41 @@ export class PrivateConsultationParticipantsComponent extends WRParticipantStatu
                     c.hearing_role !== HearingRole.WINGER
             )
             .map(c => {
-                const interpreterLink = c.linked_participants.find(x => x.link_type === LinkType.Interpreter);
-                const participant: ParticipantListItem = { ...c };
-                if (c.linked_participants && interpreterLink) {
-                    participant.interpreter = this.participantsInConsultation.find(x => x.id === interpreterLink.linked_id);
-                }
-                return participant;
+                return this.mapResponseToListItem(c);
             });
         const quickLinkParticipants = this.participantsInConsultation
             .filter(p => p.role === Role.QuickLinkParticipant)
-            .sort((a, b) => a.display_name.localeCompare(b.display_name));
+            .sort((a, b) => a.display_name.localeCompare(b.display_name))
+            .map(c => {
+                return this.mapResponseToListItem(c);
+            });
         return [...regularParticipants, ...quickLinkParticipants];
     }
 
     get johRoles(): string[] {
         return [HearingRole.JUDGE, HearingRole.PANEL_MEMBER, HearingRole.WINGER];
     }
 
-    getMemberParticipantsByRole(role: any): ParticipantListItem[] {
-        return this.participantsInConsultation.filter(p => p.hearing_role === role);
+    getMemberParticipantsByHearingRole(role: any): ParticipantListItem[] {
+        return this.participantsInConsultation
+            .filter(p => p.hearing_role === role)
+            .sort((a, b) => a.display_name.localeCompare(b.display_name))
+            .map(c => {
+                return this.mapResponseToListItem(c);
+            });
     }
 
     getWitnessesAndObservers(): ParticipantListItem[] {
         if (!this.isJohConsultation()) {
             return [];
         }
-        const witnesses = this.getMemberParticipantsByRole(HearingRole.WITNESS);
+        const witnesses = this.getMemberParticipantsByHearingRole(HearingRole.WITNESS);
         const observers = this.participantsInConsultation
             .filter(p => p.hearing_role === HearingRole.OBSERVER || p.role === Role.QuickLinkObserver)
-            .sort((a, b) => a.display_name.localeCompare(b.display_name));
+            .sort((a, b) => a.display_name.localeCompare(b.display_name))
```

Comment:
there appears to be some code duplications here with the usage of sort operators. I was wondering would be possible to create a private function here and reuse that instead, additionally I guess it will also explain the purpose of the operations. But I am happy to leave it approved if it turns out a bit complicated to do with observables.

Reply:
That's a good point. I've added a private function that handles sorting and mapping of an array. Does that make sense?